tools/network/dnssec/resolver.go

```diff
@@ -23,53 +44,63 @@ import (
 	"time"
 
 	"github.com/miekg/dns"
-)
 
-const defaultMaxHops = 10
-
-// List of DNSSEC-aware servers
-// CloudFlare: 1.1.1.1:53 1.0.0.1:53
-// Google: 8.8.8.8:53 8.8.4.4:53
-// Yandex 77.88.8.8:53 77.88.8.1:53
-// Comodo 8.26.56.26:53 8.20.247.20:53
-//
-// Other - no DNSSEC
-// OpenDNS 208.67.222.222:53
-// Baidu 180.76.76.76:53
-// Alibaba 223.6.6.6:53
+	"github.com/algorand/go-algorand/logging"
+)
 
-var defaultDnssecAwareNSServers = []string{"1.1.1.1:53", "8.8.8.8:53", "77.88.8.8:53", "8.26.56.26:53"}
+// ResolverIf represents net.Resolver-compatible interface
+type ResolverIf interface {
+	LookupAddr(ctx context.Context, addr string) (names []string, err error)
+	LookupCNAME(ctx context.Context, host string) (cname string, err error)
+	LookupHost(ctx context.Context, host string) (addrs []string, err error)
+	LookupIPAddr(ctx context.Context, host string) ([]net.IPAddr, error)
+	LookupMX(ctx context.Context, name string) ([]*net.MX, error)
+	LookupNS(ctx context.Context, name string) ([]*net.NS, error)
+	LookupPort(ctx context.Context, network, service string) (port int, err error)
+	LookupSRV(ctx context.Context, service, proto, name string) (cname string, addrs []*net.SRV, err error)
+	LookupTXT(ctx context.Context, name string) ([]string, error)
+}
 
 // Resolver provides DNSSEC resolution
 type Resolver struct {
-	resolver   netResolverIf
+	client     Querier
 	trustChain *trustChain
 	maxHops    int
 }
 
-// DefaultResolver with one DNS server and 1 second timeout
-var DefaultResolver = MakeDnssecResolver(defaultDnssecAwareNSServers, time.Second)
-
 // MakeDnssecResolver return resolver from given NS servers and timeout duration
-func MakeDnssecResolver(servers []string, timeout time.Duration) (r Resolver) {
-	rs := &resolverImpl{readTimeout: timeout, servers: servers, rootAnchor: rootAnchorXML}
+func MakeDnssecResolver(servers []ResolverAddress, timeout time.Duration) ResolverIf {
+	dc := MakeDNSClient(servers, timeout)
+	tc := &QueryWrapper{dc}
+	return &Resolver{client: dc, trustChain: makeTrustChain(tc), maxHops: DefaultMaxHops}
+}
 
-	if len(rs.servers) == 0 {
-		rs.servers = append(rs.servers, defaultDnssecAwareNSServers...)
+// MakeDefaultDnssecResolver returns a resolver with all possible DNS servers:
+// system, fallback, default
+func MakeDefaultDnssecResolver(fallbackAddress string) ResolverIf {
```

Comment:
same note regarding the logger as before.

Reply:
done